File: 1014-k-closest-points-to-origin/1014-k-closest-points-to-origin.js
/**
 * @param {number[][]} points
 * @param {number} k
 * @return {number[][]}
 */
var kClosest = function (points, k) {
    let collect_ed = []
    let minheap = new MinPriorityQueue()
    for (let i = 0; i < points.length; i++) {
        let a = points[i][0]
        let b = points[i][1]
        // Calculate ed distance
        let ed = Math.sqrt(Math.pow(a, 2) + Math.pow(b, 2))
        minheap.enqueue(points[i], ed)
    }

    while (k !== 0) {
        let el = minheap.dequeue()
        collect_ed.push(el.element)
        k--
    }
    return collect_ed
};

File: 1014-k-closest-points-to-origin/1014-k-closest-points-to-origin.py
import heapq as heap


class Solution:
    def kClosest(self, points: List[List[int]], k: int) -> List[List[int]]:
        minHeap = []
        for x, y in points:
            dist = sqrt(x**2 + y**2)
            minHeap.append((dist, [x, y]))

        heapq.heapify(minHeap)
        
        res = []
        for _ in range(k):
            dist,point = heapq.heappop(minHeap)
            res.append(point)
        return res